src/update_handler.c

```diff
@@ -748,14 +748,17 @@ static gboolean unpack_archive(RaucImage *image, gchar *dest, GError **error)
  *
  * @return TRUE on success, FALSE if an error occurred
  */
-static gboolean run_slot_hook(const gchar *hook_name, const gchar *hook_cmd, RaucImage *image, RaucSlot *slot, GError **error)
+static gboolean run_slot_hook_extra_env(const gchar *hook_name, const gchar *hook_cmd, RaucImage *image, RaucSlot *slot, GHashTable *variables, GError **error)
 {
 	g_autoptr(GSubprocessLauncher) launcher = NULL;
 	g_autoptr(GSubprocess) sproc = NULL;
 	g_autofree gchar* image_size = NULL;
 	GError *ierror = NULL;
 	gboolean res = FALSE;
 	RaucBundle *bundle;
+	GHashTableIter iter;
+	gchar *key = NULL;
+	gchar *value = NULL;
```

Comment:
This could be moved to the `if (variables)` block.

Reply:
Good point, will do that.